libtransmission/blocklist.cc

```diff
@@ -49,10 +50,67 @@ void BlocklistFile::ensureLoaded() const
         return;
     }
 
-    auto range = IPv4Range{};
-    while (in.read(reinterpret_cast<char*>(&range), sizeof(range)))
+    auto file_size = std::filesystem::file_size(filename_);
+    auto zeroes_count = 0;
+    auto max_zeroes = 0;
+
+    if (file_size >= 40)
+    {
+        std::array<char, 40> first_struct = {};
+
+        in.read(reinterpret_cast<char*>(&first_struct), 40);
+        in.clear();
+        in.seekg(0, std::ios::beg);
+
+        for (auto const struct_byte : first_struct)
+        {
+            if (struct_byte == 0)
+            {
+                zeroes_count++;
+            }
+            else
+            {
+                if (zeroes_count > max_zeroes)
+                {
+                    max_zeroes = zeroes_count;
+                }
+
+                zeroes_count = 0;
+            }
+        }
+
+        if (zeroes_count > max_zeroes)
+        {
+            max_zeroes = zeroes_count;
+        }
+    }
```

Comment:
We can fold those two `if (zeroes_count > max_zeroes)` clauses together

```suggestion
        for (auto const struct_byte : first_struct)
        {
            if (struct_byte != 0)
            {
                zeroes_count = 0;
            }
            else
            {
                ++zeroes_count;
                
                if (zeroes_count > max_zeroes)
                {
                    max_zeroes = zeroes_count;
                }
            }
        }
    }
```